woocommerce/woocommerce-android
# Introduce preview receipt UI

Parent issue #4039

Merge instructions:
1. Make sure #4326 is merged
2. Remove "Do not merge" label
2. Merge this PR

This PR adds "See receipt" button to order detail screen and introduced ReceiptPreview fragment.

Note: The PR contains over 500lines of changes, but a lot of them are tests. I recommend reviewing it by commits. 

<img src="https://user-images.githubusercontent.com/2261188/124059613-fa522d00-d9f0-11eb-8148-b46785d3a78f.png" width="250px" />
<img src="https://user-images.githubusercontent.com/2261188/124059617-fc1bf080-d9f0-11eb-9708-36fc6234693c.png" width="250px" />
<img src="https://user-images.githubusercontent.com/2261188/124059759-3d140500-d9f1-11eb-9709-90f87f473ab1.gif" width="250px" />


To test:
Prerequisites: Use card present payment ready site and order.
1. Open order detail of an unpaid order in US dollars (Notice the "See receipt" button is not visible)
2. Tap on Collect Payment
3. Complete the payment flow
4. Dismiss the "Payment successful" dialog
5. Notice the order detail screen gets updated and "See receipt" button is visible
2. Tap on "See receipt"
3. ReceiptPreview screen is shown - you should see a progress dialog while the receipt is being loaded
4. Tap on Print and notice you get redirected to the OS printing screen
5. Go back and tap on Send Email and notice an email client with pre-filled content is shown

Update release notes:
